src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,72 @@
+package duke.task;
+import java.time.LocalDateTime;
+/**
+ * Represents a deadline task as a special case of task. A <code>deadline</code> object has three
+ * fields, which are the task name, due time and done-status. e.g., <code>go to school, 2020-01-01 19:00, false</code>
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```

Comment:
Perhaps you could be clearer with the naming of this variable, maybe use `dateBy` instead of just `by`?